homeassistant/components/myio/__init__.py

```diff
@@ -0,0 +1,143 @@
+                vol.Required(CONF_USERNAME, default="admin"): str,
+                vol.Required(CONF_PASSWORD, default="admin"): str,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+COMMS_THREAD = CommsThread()
+
+
+async def async_setup(hass, config):
+    """Set up the myIO-server component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up config entry."""
+    _LOGGER.debug(DOMAIN)
+    _server_name = slugify(config_entry.data[CONF_NAME])
+    _server_status = "Offline"
+    hass.data[_server_name] = {}
+    hass.states.async_set(f"{_server_name}.state", _server_status)
+    try:
+        _refresh_timer = config_entry.options[CONF_REFRESH_TIME]
+    except (ValueError, Exception):  # pylint: disable=broad-except
+        _refresh_timer = 4
+
+    def server_status():
+        """Return the server status."""
+        return hass.states.get(f"{_server_name}.state").state
+
+    def server_data():
+        """Return the server data dictionary database."""
+        return hass.data[_server_name]
+
+    async def setup_config_entries():
+        """Set up myIO platforms with config entry."""
+
+        for component in PLATFORMS:
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(config_entry, component)
+            )
+
+        # Use `hass.async_add_job` to avoid a circular dependency
+        # between the platform and the component
+        for component in PLATFORMS:
+            hass.async_add_job(
+                hass.config_entries.async_forward_entry_setup(config_entry, component)
+            )
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        was_offline = False
+        _temp_coordinator = hass.data[_server_name]["coordinator"]
+        del hass.data[_server_name]["coordinator"]
+
+        if server_status() == "Offline":
+            hass.states.async_set(f"{_server_name}.available", False)
+            was_offline = True
+            for component in PLATFORMS:
+                await hass.config_entries.async_forward_entry_unload(
+                    config_entry, component
+                )
+
+        [hass.data[_server_name], _server_status] = await COMMS_THREAD.send(
+            server_data=server_data(),
+            server_status=server_status(),
+            config_entry=config_entry,
+            _post=None,
+        )
+
+        hass.states.async_set(f"{_server_name}.state", _server_status)
+
+        if server_status().startswith("Online") and was_offline:
+            hass.states.async_set(f"{_server_name}.available", True)
+            await setup_config_entries()
+            _LOGGER.debug("Online")
+        elif not was_offline and server_status() == "Offline":
+            _LOGGER.debug("PlatformNotReady")
+            hass.data[_server_name]["coordinator"] = _temp_coordinator
+            raise PlatformNotReady
```

Comment:
This should raise `ConfigEntryNotReady`, and why not raise it directly in the beginning of the entry setup?

Reply:
this is a part of async_update_data() ... this is started with the coordinator.... this is what refreshing the whole database (json)... and checks, if the server goes offline...

Reply:
The DataUpdateCoordinator does these checks already?

Reply:
this script is runing in the coordinator... this part of code runing immediately after the pull fresh data from myIO server... (COMMS_THREAD)... this is the only way to get information from myIO server...